### ISSUE TITLE
New cop: Style/MagicNumber

### ISSUE THREAD

**Issue by fatkodima** [rubocop-hq/rubocop#8460]
Similar checks for other languages linters: https://palantir.github.io/tslint/rules/no-magic-numbers/ and https://checkstyle.sourceforge.io/config_coding.html#MagicNumber

Raw numbers would be only allowed in different kinds of assignments and from explicit list of numbers, including `-1`, `0` and `1` by default.

@bbatsov Thoughts on this?

**Comment by marcandre** [comment]
It's a good rule of thumb. In a few contexts though that rule would be problematic; should we lookout for these?
Off the top of my head:

```ruby
enum.each_const(2) do |prev, next| # ...
enum.each_const(3) do |prev, cur, next| # ...

if proc.arity == 2 # or other values ...
```

It would be interesting to see on our codebase if there are many offenses...

**Comment by fatkodima** [comment]
Ok, I have implemented a simple proof of concept - and yes, there are many valid cases, but it generates a lot of false positives and it is very hard/impossible to catch them all. So I agree that 

> It's a good rule of thumb.

and it would be better to leave it so and do not implement any cops regarding this.

@marcandre Do you agree?

**Comment by bbatsov** [comment]
> ut it generates a lot of false positives and it is very hard/impossible to catch them all

That's why we dumped this idea in the past. It's a great guideline, but it's really hard to get it right. Alternatively, we can have a simple cop that's disabled and people are just expected to run it with `--only` from time to time. Still, it seems to me we can spent our time better elsewhere. :-)

**Comment by fatkodima** [comment]
Agreed.
